consensus/src/round_manager.rs

```diff
@@ -531,8 +560,34 @@ impl RoundManager {
         );
 
         observe_block(proposal.timestamp_usecs(), BlockStage::SYNCED);
+        self.process_verified_proposal(proposal).await
+    }
 
+    async fn resend_verified_proposal_to_self(&self, proposal: Block, interval_ms: u64) {
+        let author = self.network.author();
+        let self_sender = self.round_manager_tx.clone();
+        let event = VerifiedEvent::VerifiedProposalMsg(Box::new(proposal));
+        tokio::spawn(async move {
+            sleep(Duration::from_millis(interval_ms)).await;
+            if let Err(e) = self_sender.push((author, discriminant(&event)), (author, event)) {
+                error!("Failed to send event to round manager {:?}", e);
+            }
+        });
+    }
+
+    pub async fn process_verified_proposal(&mut self, proposal: Block) -> Result<()> {
```

Comment:
we can rename it back to process_proposal I think since it doesn't go through the network task now

Reply:
There is already a `process_proposal` function - I think `process_verified_proposal` proposal is a good name as we are skipping a bunch of verification done in `process_proposal`.  I am open to renaming it if you have better suggestion.